Adapter_Revit_UI/CRUD/Read.cs

```diff
@@ -59,147 +61,115 @@ protected override IEnumerable<IBHoMObject> IRead(Type type, IList ids, ActionCo
                 return null;
             }
 
-            List<FilterRequest> filterRequestList = new List<FilterRequest>();
-            if (ids != null && ids.Count > 0)
+            List<int> elementIds = new List<int>();
+            List<string> uniqueIds = new List<string>();
+            if (ids != null)
             {
-                List<string> uniqueIDList = new List<string>();
-                List<int> elementIDList = new List<int>();
                 foreach (object obj in ids)
-                    if(obj != null)
+                {
+                    if (obj is int)
+                        elementIds.Add((int)obj);
+                    else if (obj is string)
                     {
-                        if (obj is int)
-                            elementIDList.Add((int)obj);
-                        else if (obj is string)
-                            uniqueIDList.Add((string)obj);
+                        string stringId = (string)obj;
+                        int id;
+                        if (int.TryParse(stringId, out id))
+                            elementIds.Add(id);
+                        else
+                            uniqueIds.Add(stringId);
                     }
-
-                FilterRequest filterRequestUniqueIDs = null;
-                FilterRequest filterRequestElementIDs = null;
-
-                if (uniqueIDList.Count > 0)
-                    filterRequestUniqueIDs = BH.Engine.Adapters.Revit.Create.SelectionFilterRequest(uniqueIDList);
-
-                if (elementIDList.Count > 0)
-                    filterRequestElementIDs = BH.Engine.Adapters.Revit.Create.SelectionFilterRequest(elementIDList);
-
-                if (filterRequestUniqueIDs != null && filterRequestElementIDs != null)
-                {
-                    filterRequestList.Add(BH.Engine.Adapters.Revit.Create.LogicalOrFilterRequest(new List<FilterRequest>() { filterRequestElementIDs, filterRequestUniqueIDs }));
                 }
-                else
-                {
-                    if(filterRequestUniqueIDs != null)
-                        filterRequestList.Add(filterRequestUniqueIDs);
-
-                    if (filterRequestElementIDs != null)
-                        filterRequestList.Add(filterRequestElementIDs);
-                }
-
             }
 
-            if(type != null)
+            if (elementIds.Count == 0 && uniqueIds.Count == 0)
+                return Read(new FilterRequest() { Type = type } as IRequest, actionConfig);
```

Comment:
The base adapter does already a similar thing in the Pull. If you specify only a Type in the request input, it will convert it into a FilterRequest with that type.
I don't know if you knew that. This is additional and does not hurt, but consider deleting it (or replacing it with an error message, like "no ids specified") if the other method works and you think is enough for you. 
Otherwise, leave this (no problem for me).

Reply:
@alelom I had an issue with infinite loop here as far as I remember - this line (particularly casting to `IRequest` in order to call `BH.UI.Revit.Adapter.RevitUIAdapter.Read` instead of `BH.Adapter.BHoMAdapter.Read`) is meant to avoid it.

Reply:
Strange, I can't see neither how the infinite loop might happen, nor how this would solve that. Ok anyway